src/mudlet.cpp

```diff
@@ -2335,37 +2331,53 @@ void mudlet::slot_replaySpeedDown()
     replaySpeedDisplay->show();
 }
 
+/* loop through and stop all sounds */
 void mudlet::stopSounds()
 {
-    mpMusicBox1->stop();
-    mpMusicBox2->stop();
-    mpMusicBox3->stop();
-    mpMusicBox4->stop();
+    QListIterator<QMediaPlayer *> itMusicBox( MusicBoxList );
+    
+    while( itMusicBox.hasNext() ) {
+        itMusicBox.next()->stop();
+    }
 }
 
-void mudlet::playSound( QString s )
+void mudlet::playSound( QString s, int soundVolume )
 {
-    if( mpMusicBox1->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox1->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox1->play();
-    }
-    else if( mpMusicBox2->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox2->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox2->play();
-    }
-    else if( mpMusicBox3->state() != QMediaPlayer::PlayingState )
-    {
-        mpMusicBox3->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox3->play();
+    QListIterator<QMediaPlayer *> itMusicBox( MusicBoxList );
+    QMediaPlayer * pPlayer=0;
+    
+    /* find first available inactive QMediaPlayer */
+    while( itMusicBox.hasNext() ) {
+        QMediaPlayer * pTestPlayer=itMusicBox.next();
+        
+        if ( pTestPlayer->state() != QMediaPlayer::PlayingState && pTestPlayer->mediaStatus() != QMediaPlayer::LoadingMedia ) {
+            pPlayer=pTestPlayer;
+            break;
+        }
     }
-    else
-    {
-        mpMusicBox4->stop();
-        mpMusicBox4->setMedia( QUrl::fromLocalFile( s ) );
-        mpMusicBox4->play();
+
+    /* no available QMediaPlayer, create a new one */
+    if (!pPlayer) {
+        pPlayer = new QMediaPlayer(this);
+
+        if (!pPlayer) {
+            /* It (should) be impossible to ever reach this */
+            Host * pH = getActiveHost();
+            
+            if (pH) {
+                pH->postMessage( "\n[  ERROR  ]  - Unable to create new QMediaPlayer object\n" );    
```

Comment:
Take a look though the map related classs e.g. `TMap`, `TRoomDB`, `TArea`, `TRoom` {which is an area where I have been particularly active :blush: } besides which, as you have poked around in the `cTelnet::postMessage(...)` you might see that the code looks for the `"[ something ]    - "` tag on the __first__ line.  FTR once the code detects this type of string the code aligns all following lines to the same indentation as the position of the first character after the ending `"... - "` of the prefix so I try to ensure that all messages have the same width in the portion `"[ something ] - "` so that they all align on screen in the console; you may notice the in source code formatting of such messages try to replicate this alignment.  I also try to avoid individual message lines including indentation to be over 80 characters so that they don't trip the line wrapping that the main console text display stuff does.  This doesn't always work - especially when you have `%1`...`%10` type variable replacements by strings...